highlight/highlight
# Fix search params being set in wrong place

## Summary

We have a bug where interacting with UI in the related resource panel will manipulate the main page's search query rather than the query used for the related resource in the panel. e.g. if you are on the traces page and click on "Apply as filter" on a log attribute from the logs' related resource panel it will apply the filter on the traces' query.

My workaround for this is to move some of the search state/logic into a `SearchContext` that can be added around search components, then use the context (via `useSearchContext`) to access a setter that will update the `query` just for that instance of the search.

Also makes a few small tweaks to fix some issues I noticed while click testing.

* Centers the reset ("X") button to the right of the search filters
* Sorts the root spans so we always show the longest one first if we have two that start around the same time (temporary fix improvement until we fix our timing issues)

## How did you test this change?

* Go to the traces page
* Open a trace
* Click "View logs" on the trace panel
* Expand a log and click "Apply as filter" on one of the attributes
  * Make sure this doesn't update the traces' search query

## Are there any deployment considerations?

Should only fix a client-side bug. No backend changes.

## Does this work require review from our design team?

N/A
